x-pack/plugins/reporting/server/export_types/printable_pdf/lib/pdf/index.ts

```diff
@@ -0,0 +1,148 @@
+  private _logo: string | undefined;
+  private _title: string;
+  private _content: Content[];
+  private _printer: Printer;
+  private _pdfDoc: PDFKit.PDFDocument | undefined;
+
+  constructor(layout: LayoutInstance, logo: string | undefined) {
+    const fontPath = (filename: string) => path.resolve(assetPath, 'fonts', filename);
+    const fonts = {
+      Roboto: {
+        normal: fontPath('roboto/Roboto-Regular.ttf'),
+        bold: fontPath('roboto/Roboto-Medium.ttf'),
+        italics: fontPath('roboto/Roboto-Italic.ttf'),
+        bolditalics: fontPath('roboto/Roboto-Italic.ttf'),
+      },
+      'noto-cjk': {
+        // Roboto does not support CJK characters, so we'll fall back on this font if we detect them.
+        normal: fontPath('noto/NotoSansCJKtc-Regular.ttf'),
+        bold: fontPath('noto/NotoSansCJKtc-Medium.ttf'),
+        italics: fontPath('noto/NotoSansCJKtc-Regular.ttf'),
+        bolditalics: fontPath('noto/NotoSansCJKtc-Medium.ttf'),
+      },
+    };
+
+    this._layout = layout;
+    this._logo = logo;
+    this._title = '';
+    this._content = [];
+    this._printer = new Printer(fonts);
+  }
+
+  _addContents(contents: Content[]) {
+    const groupCount = this._content.length;
+
+    // inject a page break for every 2 groups on the page
+    if (groupCount > 0 && groupCount % this._layout.groupCount === 0) {
+      contents = [
+        ({
+          text: '',
+          pageBreak: 'after',
+        } as ContentText) as Content,
+      ].concat(contents);
+    }
+    this._content.push(contents);
+  }
+
+  addImage(base64EncodedData: string, { title = '', description = '' }) {
+    const contents: Content[] = [];
+
+    if (title && title.length > 0) {
+      contents.push({
+        text: title,
+        style: 'heading',
+        font: getFont(title),
+        noWrap: true,
+      });
+    }
+
+    if (description && description.length > 0) {
+      contents.push({
+        text: description,
+        style: 'subheading',
+        font: getFont(description),
+        noWrap: true,
+      });
+    }
+
+    const size = this._layout.getPdfImageSize();
+    const img = {
+      image: `data:image/png;base64,${base64EncodedData}`,
+      alignment: 'center',
+      height: size.height,
+      width: size.width,
+    };
+
+    const wrappedImg = {
+      table: {
+        body: [[img]],
+      },
+      layout: 'noBorder',
+    };
+
+    contents.push(wrappedImg);
+
+    this._addContents(contents);
+  }
+
+  setTitle(title: string) {
+    this._title = title;
+  }
+
+  generate() {
+    const docTemplate = _.assign(
+      getTemplate(this._layout, this._logo, this._title, tableBorderWidth, assetPath),
+      {
+        content: this._content,
+      }
+    );
+    this._pdfDoc = this._printer.createPdfKitDocument(docTemplate, getDocOptions(tableBorderWidth));
+    return this;
+  }
+
+  getBuffer(): Promise<Buffer | null> {
+    return new Promise((resolve, reject) => {
+      if (!this._pdfDoc) {
```

Comment:
Note: the previous code did the undefined check outside of the Promise, which caused TS to complain about lines 143-145

https://github.com/elastic/kibana/blob/a74cea30c55e828ac5925f6daa083172c8d8f67d/x-pack/plugins/reporting/server/export_types/printable_pdf/lib/pdf/index.js#L138